### PR TITLE
Mobile navigation pt. 2/2

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #6389 

## Description of Changes
- removed legacy code for mobile menu
- refactored code related to sublayout header (moved around, changed names, simplified)
- added search page to mobile top navbar
- added right side user drawer to mobile top navbar  
- refactored components related to UserDropdown so the menu content can be used in destkop and mobile components

## Test Plan
- click around on mobile device or in chrome dev tools
- new navigation includes 
  - left sidebar (scoped/non-scoped pages)
  - top navbar 
  - search icon + full page
  - right side user drawer
  - bottom navigation  

https://github.com/hicommonwealth/commonwealth/assets/14819225/ab35cf5d-4669-4b06-b9e6-355c3d0c238c

## Deployment Plan

1. safe to deploy

### CODE REVIEW

**Comment by jnaviask** [comment]
Let's try to get this in for v1.0.

**Comment by mzparacha** [comment]
Testing locally works great, just a minor issue on ~ `800px`, the app sidebar toggle button is missing

image

Going through the code.....

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx [hunk 1]
```diff
@@ -82,7 +94,7 @@ export const CWSearchBar: FC<SearchBarProps> = ({
     // fire before resetting the search bar
     if (!resetTimer) {
       resetTimer = setTimeout(() => {
-        resetSearchBar();
+        // resetSearchBar();
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx:97]: maybe remove?

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx [hunk 2]
```diff
@@ -64,6 +70,12 @@ export const CWSearchBar: FC<SearchBarProps> = ({
     'members',
   ]);
 
+  useEffect(() => {
+    if (size === 'small') {
+      inputRef?.current?.focus?.();
+    }
+  }, [size]);
+
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx:78]: now sure why we need to do this, do we ever change `size` prop based on screen size?

**Review by mzparacha — APPROVED** [review]
Awesome work @masvelio 🔥

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx:97]: no no, this should be back, thanks for pointing out

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx:78]: so the search component is complicated and it definitely needs refactor but first it needs rethink for desktop and mobile UX. Right know I used current searchbar component and just adjusted it to work on mobile. Yes, on mobile we use size=small and this useEffect makes input to focus when the modal appears

https://github.com/hicommonwealth/commonwealth/assets/14819225/07f246a9-7b2c-4a3d-9eb1-98b3b6ad998a

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar/CWSearchBar.tsx:79]: That makes sense, thanks!

**Pull request merged**